server/src/test/java/org/opensearch/extensions/rest/RestInitializeExtensionActionTests.java

```diff
@@ -121,4 +130,94 @@ public void testRestInitializeExtensionActionFailure() throws Exception {
         );
     }
 
+    public void testRestInitializeExtensionActionResponseWithAdditionalSettings() throws Exception {
+        Setting boolSetting = Setting.boolSetting("boolSetting", false, Setting.Property.ExtensionScope);
+        Setting stringSetting = Setting.simpleString("stringSetting", "default", Setting.Property.ExtensionScope);
+        Setting intSetting = Setting.intSetting("intSetting", 0, Setting.Property.ExtensionScope);
+        Setting listSetting = Setting.listSetting(
+            "listSetting",
+            List.of("first", "second", "third"),
+            Function.identity(),
+            Setting.Property.ExtensionScope
+        );
+        ExtensionsManager extensionsManager = new ExtensionsManager(Set.of(boolSetting, stringSetting, intSetting, listSetting));
+        ExtensionsManager spy = spy(extensionsManager);
+
+        // optionally, you can stub out some methods:
+        when(spy.getAdditionalSettings()).thenCallRealMethod();
+        Mockito.doCallRealMethod().when(spy).loadExtension(any(ExtensionsSettings.Extension.class));
+        Mockito.doNothing().when(spy).initialize();
+        RestInitializeExtensionAction restInitializeExtensionAction = new RestInitializeExtensionAction(spy);
+        final String content = "{\"name\":\"ad-extension\",\"uniqueId\":\"ad-extension\",\"hostAddress\":\"127.0.0.1\","
+            + "\"port\":\"4532\",\"version\":\"1.0\",\"opensearchVersion\":\""
+            + Version.CURRENT.toString()
+            + "\","
+            + "\"minimumCompatibleVersion\":\"3.0.0\",\"boolSetting\":true,\"stringSetting\":\"customSetting\",\"intSetting\":5,\"listSetting\":[\"one\",\"two\",\"three\"]}";
```

Comment:
Just FYI: You will need to change the `minimumCompatibleVersion` to `2.0.0` while backporting to 2.x or else it will the validate check [here](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/extensions/DiscoveryExtensionNode.java#L106).

Reply:
Is there a better way to set minimumCompatibleVersion to reference the current major version?

Reply:
You can probably pull the version from `Version.CURRENT.minimumCompatibilityVersion()`.

Reply:
Thanks for the suggestion :). I removed all hardcoded versions. 🤞 the backport is created automatically.